src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
-
-				for( BakedQuad bakedQuad : blockModel.getQuads( state, facing, rand ) )
+				if( !blockState.getBlock().canRenderInLayer( blockState, layer ) )
 				{
-					return new TextureAtlasAndTint( bakedQuad );
+					continue;
 				}
+			}
 
-				for( BakedQuad bakedQuad : blockModel.getQuads( state, null, rand ) )
+			AxisAlignedBB fullBounds = thinFacades ? THIN_FACADE_BOXES[sideIndex] : THICK_FACADE_BOXES[sideIndex];
+			AxisAlignedBB facadeBox = fullBounds;
+			//If we are a transparent facade, we need to modify out BB.
+			if( facadeRenderState.isTransparent() )
+			{
+				double offset = thinFacades ? THIN_THICKNESS : THICK_THICKNESS;
+				AEAxisAlignedBB tmpBB = null;
+				for( EnumFacing face : EnumFacing.VALUES )
 				{
-					if( firstFound == null )
-					{
-						firstFound = new TextureAtlasAndTint( bakedQuad );
-					}
-					if( bakedQuad.getFace() == facing )
+					//Only faces that aren't on our axis
+					if( face.getAxis() != side.getAxis() )
 					{
-						return new TextureAtlasAndTint( bakedQuad );
+						FacadeRenderState otherState = facadeStates.get( face );
+						if( otherState != null && !otherState.isTransparent() )
+						{
+							if( tmpBB == null )
+							{
+								tmpBB = AEAxisAlignedBB.fromBounds( facadeBox );
+							}
+							switch( face )
+							{
+								case DOWN:
+									tmpBB.minY += offset;
+									break;
+								case UP:
+									tmpBB.maxY -= offset;
+									break;
+								case NORTH:
+									tmpBB.minZ += offset;
+									break;
+								case SOUTH:
+									tmpBB.maxZ -= offset;
+									break;
+								case WEST:
+									tmpBB.minX += offset;
+									break;
+								case EAST:
+									tmpBB.maxX -= offset;
+									break;
+								default:
+									throw new RuntimeException( "Switch falloff. " + String.valueOf( face ) );
+							}
+						}
 					}
 				}
+				if( tmpBB != null )
+				{
+					facadeBox = tmpBB.getBoundingBox();
+				}
 			}
-		}
-		catch( Exception e )
-		{
-			if( warnedFor.add( state.getBlock().getRegistryName() ) )
-			{
-				AELog.warn( "Unable to get facade sprite for blockstate %s. Supressing further warnings for this block.", state );
-				AELog.debug( e );
-			}
-		}
-		finally
-		{
-			ForgeHooksClient.setRenderLayer( orgLayer );
-		}
 
-		// Fall back to the particle texture, if we havent found anything else so far.
-		if( firstFound == null )
-		{
+			AEAxisAlignedBB cutOutBox = getCutOutBox( facadeBox, partBoxes );
+			List<AxisAlignedBB> holeStrips = getBoxes( facadeBox, cutOutBox, side.getAxis() );
+			IBlockAccess facadeAccess = new FacadeBlockAccess( parentWorld, pos, side, blockState );
+
+			BlockRendererDispatcher dispatcher = Minecraft.getMinecraft().getBlockRendererDispatcher();
+
 			try
 			{
-				return new TextureAtlasAndTint( blockModel.getParticleTexture(), -1 );
+				blockState = blockState.getActualState( facadeAccess, pos );
 			}
-			catch( Exception e )
+			catch( Exception ignored )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)